### PR TITLE
FOPTS-3024 - New Budget vs Actual Spend report policy

### PR DESCRIPTION
**Pull request by night-codes**
### Description

Email a report of budget vs actual spend so the customer doesn't need to login to Flexera One

### Issues Resolved

https://flexera.atlassian.net/browse/FLEX-204

### Link to Example Applied Policy

Full budget report:
https://app.flexera.com/orgs/1105/automation/applied-policies/projects/60073?policyId=65e0c8b01b78750001a4d4ab

With filtered dimensions:
https://app.flexera.com/orgs/1105/automation/incidents/projects/60073?incidentId=65e21c3d99c5e4000120eb4e

[Budget used (to compare data and chart)](https://app.flexera.com/orgs/1105/optima/budgets/1ef4c91?budgetName=FinOps%202023%202&endDate=2023-12&startDate=2023-01) 

### Contribution Check List

- [ ] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: cost/budget_v_actual_spend_report/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Budget vs Actual Spend Policy
```
Comment (TDubovchenko) [inline cost/budget_v_actual_spend_report/README.md:1]: Policy name is "Budget vs Actual Spend Report"

File: cost/budget_v_actual_spend_report/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+## Functional Details
+
+- This policy allows dynamic configuration of `activeDeadlineSeconds` via `values.yaml`, enabling users to set the maximum duration for cron job execution.
+- Chart templates are updated for improved configuration adaptability.
+- Various minor enhancements and bug fixes contribute to improved stability and performance.
+
+## Input Parameters
+
+- _Budget Name or ID_: The name or ID of the target Budget.
+- _Filter Group By Dimensions_: Filter by dimension=value pairs (e.g., 'Cloud Vendor=AWS'). Multiple values for the same dimension can be comma-separated.
```
Comment (TDubovchenko) [inline cost/budget_v_actual_spend_report/README.md:27]: Multiple values for the same dimension can be supplied as coma-separated list.

File: cost/budget_v_actual_spend_report/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+- _Budget Alert Type_: Options are "Actual" or "Forecasted". Actual Spend alerts are based on incurred costs, while Forecasted Spend alerts use monthly runrates.
```
Comment (TDubovchenko) [inline cost/budget_v_actual_spend_report/README.md:28]: This parameter is not available

File: cost/budget_v_actual_spend_report/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+- [**Flexera Credential**](https://docs.flexera.com/flexera/EN/Automation/ProviderCredentials.htm) (_provider=flexera_) which has the following roles:
+  - `optima:budget:index`
+  - `optima:budget:report`
+  - `optima:billing_center:show`
+
+Refer to the [Provider-Specific Credentials](https://docs.flexera.com/flexera/EN/Automation/ProviderCredentials.htm) page for detailed instructions on setting up Credentials.
+
+## Functional Details
+
+- This policy allows dynamic configuration of `activeDeadlineSeconds` via `values.yaml`, enabling users to set the maximum duration for cron job execution.
```
[the remark this answers is not in the document]
Comment (night-codes) [reply cost/budget_v_actual_spend_report/README.md:20]: Ooops. No, this is a mock description that I forgot to remove. Will do it

File: cost/budget_v_actual/monthly_budget_v_actual.pt [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
 type "policy"
-short_description "This policy allows you to set up scheduled reports that will provide monthly actual v. budgeted cloud cost across all resources in the Billing Center(s) you specify, delivered to any email addresses you specify. See the [README](https://github.com/flexera-public/policy_templates/tree/master/cost/budget_v_actual) and [docs.flexera.com/flexera/EN/Automation](https://docs.flexera.com/flexera/EN/Automation/AutomationGS.htm) to learn more."
+short_description "**Deprecated: This policy is no longer being updated.**  This policy allows you to set up scheduled reports that will provide monthly actual v. budgeted cloud cost across all resources in the Billing Center(s) you specify, delivered to any email addresses you specify. See the [README](https://github.com/flexera-public/policy_templates/tree/master/cost/budget_v_actual) and [docs.flexera.com/flexera/EN/Automation](https://docs.flexera.com/flexera/EN/Automation/AutomationGS.htm) to learn more."
 severity "low"
 category "Cost"
 tenancy "single"
 default_frequency "daily"
 info(
-  version: "2.0",
+  version: "2.1",
```
[the remark this answers is not in the document]
Comment (night-codes) [reply cost/budget_v_actual/monthly_budget_v_actual.pt:10]: Added

File: cost/budget_v_actual_spend_report/budget_v_actual_spend_report.pt [hunk 6]
```diff
@@ -0,0 +1,499 @@
...
+    header "Api-Version", "0.1"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    field "id", jmes_path(response,"id")
+    field "value", jmes_path(response,"value")
+  end
+end
+
```
Comment (juanestrahcl) [inline cost/budget_v_actual_spend_report/budget_v_actual_spend_report.pt:113]: Remove blank line.

File: cost/budget_v_actual_spend_report/budget_v_actual_spend_report.pt [hunk 7]
```diff
@@ -0,0 +1,499 @@
...
+      _.each(b.budgetYearMonths, function (yearMonth) {
+        var curr = currDate.getFullYear() + '-' + ('0' + (currDate.getMonth()+1)).slice(-2);
+        b.start_date = Math.min(b.start_date, Date.parse(yearMonth))
+        b.end_date = Math.max(b.end_date, Date.parse(yearMonth))
+      })
+      return (b.name == budget_name_id || b.id == budget_name_id);
+    });
+EOS
+end
+
```
Comment (juanestrahcl) [inline cost/budget_v_actual_spend_report/budget_v_actual_spend_report.pt:213]: Remove blank line.

File: cost/budget_v_actual_spend_report/budget_v_actual_spend_report.pt [hunk 8]
```diff
@@ -0,0 +1,499 @@
...
+			"endAt": fmtDate(endAt.getTime()),
+    },
+    headers: {
+      "User-Agent": "RS Policies",
+      "Api-Version": "1.0"
+    }
+  }
+  EOS
+end
+
```
Comment (juanestrahcl) [inline cost/budget_v_actual_spend_report/budget_v_actual_spend_report.pt:266]: Remove blank line.

File: cost/budget_v_actual_spend_report/budget_v_actual_spend_report.pt [hunk 9]
```diff
@@ -0,0 +1,499 @@
...
+      if(i === 0) {
+        result += monthNames[monthIndex] + " " + year;
+      } else {
+        result += "|" + monthNames[monthIndex];
+      }
+    }
+
+    return result;
+  }
+
```
Comment (juanestrahcl) [inline cost/budget_v_actual_spend_report/budget_v_actual_spend_report.pt:331]: Remove blank line.

**Review by juanestrahcl — APPROVED** [review]

**Review by juanestrahcl — APPROVED** [review]

**Review by TDubovchenko — APPROVED** [review]

**Pull request merged**